### ISSUE TITLE
Module build failed: SyntaxError: Unexpected token (8:4) while using typescript and styled-components

### ISSUE THREAD

**Issue by ColinPNtenga** [storybooks/storybook#4032]
I got following error while trying to use storybook with typescript and styled-components. I have tried all solutions provided in previous issues without any luck. 

```
`WARNING in ./src/components/Button/story.tsx
Module build failed: SyntaxError: /Users/colnteng/Desktop/meglo-siteapp-typescript/src/components/Button/story.tsx: Unexpected token (7:4)

  5 | storiesOf('components/Button', module)
  6 |   .add('with text', () => (
> 7 |     <Button label={'Hello there!'} onClick={() => alert('It worked!')}/>
    |     ^
  8 |   ))
`
```
Here is my .babelrc file in .storybook folder 
```
{
  "presets": [
    "env",
    "react"
  ]
}
```

Here is my webpack.config.js file in .storybook folder
```
const genDefaultConfig = require("@storybook/react/dist/server/config/defaults/webpack.config.js");

const path = require("path");

module.exports = (baseConfig, env) => {
  const config = genDefaultConfig(baseConfig, env);

  config.resolve.extensions.push(".ts", ".tsx");

  config.module.rules[0].test = /\.(ts|tsx)$/;
  config.module.rules[0].query.presets = ["@babel/preset-env"];

  config.module.rules.unshift({
    test: /\.(ts|tsx)$/,
    loader: require.resolve("ts-loader"),
    include: [path.resolve(__dirname, "../src")]
  });

  // [ts-loader, babel-loader, ...]

  return config;
};
```
Here is my package.json file 
```
"scripts": {
    "dev": "next",
    "build": "next build",
    "start": "next start",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "build-storybook": "build-storybook"
  },
  "dependencies": {
    "@zeit/next-typescript": "^1.1.0",
    "next": "^6.1.1",
    "react": "^16.4.2",
    "react-dom": "^16.4.2",
    "styled-components": "^3.3.3"
  },
"devDependencies": {
    "@babel/core": "^7.0.0-rc.1",
    "@babel/preset-env": "^7.0.0-rc.1",
    "@babel/preset-react": "^7.0.0-rc.1",
    "@storybook/addon-actions": "^3.3.12",
    "@storybook/addon-links": "^3.3.12",
    "@storybook/addons": "^3.3.12",
    "@storybook/react": "^3.3.12",
    "@types/next": "^6.1.2",
    "@types/react": "^16.4.11",
    "@types/storybook__react": "^3.0.9",
    "babel-core": "^7.0.0-bridge.0",
    "babel-loader": "^8.0.0-beta.2",
    "babel-plugin-styled-components": "^1.5.1",
    "ts-loader": "^4.4.2",
    "typescript": "^2.4.2"
  }
```
Here is my tsconfig.json
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "jsx": "preserve",
    "allowJs": true,
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "removeComments": false,
    "preserveConstEnums": true,
    "sourceMap": true,
    "skipLibCheck": true,
    "baseUrl": ".",
    "typeRoots": [
      "./node_modules/@types"
    ],
    "lib": [
      "dom",
      "es2015",
      "es2016"
    ]
  }
}
```

**Comment by igor-dv** [comment]
Did you see this guide - https://storybook.js.org/configurations/typescript-config/ ?

**Comment by Inlesco** [comment]
@ColinPNtenga ColinPNtenga I had a similar issue recently, Storybooks unable to compile JSX templates. My setup was without TS and styled-components, though.

The solution was to remove `.babelrc` config file from `.storybooks` directory as I had created it there while trying to solve Webpack's `require.context` issue. Solved both, in the end.

As for the solution of `.babelrc` config, it looks like it was definitely overriding the defaults of the Storybooks.
